portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -96,12 +96,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     List<String> comments = new ArrayList<>();
     for (Entity comment: results.asIterable()) {
-      comments.add((String)comment.getProperty("message"));
-
-      -- limit;
       if (limit == 0) {
         break;
       }
+      -- limit;
+      
+      comments.add((String)comment.getProperty("message"));
```

Comment:
These changes seem unrelated? Refrain from bundling together unrelated changes (I understand the temptation).
There's nothing wrong to having a very small PR for a quick bugfix.